File: Go/unique_email_addresses/929_test.go
package uniqueemailaddresses

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_numUniqueEmails(t *testing.T) {
	tests := []struct {
		emails []string
		want   int
	}{
		{
			emails: []string{"test.email+alex@leetcode.com", "test.e.mail+bob.cathy@leetcode.com", "testemail+david@lee.tcode.com"},
			want:   2,
		},
		{
			emails: []string{"a@leetcode.com", "b@leetcode.com", "c@leetcode.com"},
			want:   3,
		},
	}
	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := numUniqueEmails(tt.emails)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/unique_emails/929.test.ts
import { numUniqueEmails, numUniqueEmails2 } from "./929";

interface tt {
  emails: string[];
  want: number;
}

const tests: tt[] = [
  {
    emails: [
      "test.email+alex@leetcode.com",
      "test.e.mail+bob.cathy@leetcode.com",
      "testemail+david@lee.tcode.com",
    ],
    want: 2,
  },
  {
    emails: ["a@leetcode.com", "b@leetcode.com", "c@leetcode.com"],
    want: 3,
  },
];

describe("unique emails address", () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = numUniqueEmails(t.emails);
      const result2 = numUniqueEmails2(t.emails);
      expect(result).toBe(t.want);
      expect(result2).toBe(t.want);
    });
  });
});
